kimchi/src/prover.rs

```diff
@@ -711,17 +747,31 @@ where
                         .iter()
                         .map(|c| c.eval(e, index.max_poly_size))
                         .collect(),
-                    table: lcs
-                        .lookup_table
-                        .iter()
-                        .map(|p| p.eval(e, index.max_poly_size))
-                        .rev()
-                        .fold(vec![Fr::<G>::zero()], |acc, x| {
-                            acc.into_iter()
-                                .zip(x.iter())
-                                .map(|(acc, x)| acc * joint_combiner + x)
-                                .collect()
-                        }),
+                    table: {
+                        let base_table = lcs
+                            .lookup_table
+                            .iter()
+                            .map(|p| p.eval(e, index.max_poly_size))
+                            .rev()
+                            .fold(vec![Fr::<G>::zero()], |acc, x| {
+                                acc.into_iter()
+                                    .zip(x.iter())
+                                    .map(|(acc, x)| acc * joint_combiner + x)
+                                    .collect()
+                            });
+                        match lcs.table_ids.as_ref() {
+                            None => base_table,
+                            Some(table_ids) => {
+                                let table_combiner =
+                                    joint_combiner.pow([lcs.configuration.max_joint_size as u64]);
+                                base_table
+                                    .into_iter()
+                                    .zip(table_ids.eval(e, index.max_poly_size))
```

Comment:
btw this has now changed https://github.com/o1-labs/proof-systems/pull/467/

```rust
to_chunked_polynomial(index.max_poly_size).evaluate_chunks(e)
```

that being said, do we really want to handle split polynomials this way for lookup tables (and table_ids)? As we will need to cleanly ignore specific rows (the zk rows). More generally I feel like it'd be better to just deal with evaluations instead of chunked evaluations and treat the quotient polynomial (I believe the only exception) as an exception with a special type or something